components/automate-ui/src/app/modules/infra-proxy/infra-environment-constraint/infra-environment-constraint.component.ts

```diff
@@ -0,0 +1,148 @@
+import {
+  Component,
+  OnInit,
+  Input,
+  Output,
+  EventEmitter
+} from '@angular/core';
+
+import { FormGroup } from '@angular/forms';
+
+import { CookbookConstraintGrid } from '../create-environment-modal/create-environment-modal.component';
+
+@Component({
+  selector: 'app-infra-environment-constraint',
+  templateUrl: './infra-environment-constraint.component.html',
+  styleUrls: ['./infra-environment-constraint.component.scss']
+})
+
+export class InfraEnvironmentConstraintComponent implements OnInit {
+
+  @Input() constraintKeys: string[] = [];
+  @Input() cookbookConstraints: Array<CookbookConstraintGrid> = [];
+  @Input() name_id: string;
+  @Input() nameKeys: string[] = [];
+  @Input() selectedCookbookNames: string[] = [];
+  @Input() constraintFormGroup: FormGroup;
+  @Output() constraintValues: EventEmitter<CookbookConstraintGrid[]> = new EventEmitter();
+
+  public conflictError = false;
+  public operator_id = '';
+  public operatorKeys: string[] = [];
+
+  constructor(
+  ) {
+
+    this.operatorKeys = ['~>', // approximately greater than;
+    // also known as "pessimistically greater than", or "pessimistic"
```

Comment:
I realize you copied this from the docs, but it is still a mystery.
Your job is to explain your code to the reader. What does it mean? This comment is inadequate.

Also, would suggest to rename the variable to `operators` instead of `operatorKeys`, to be less confusing.